core/src/main/java/io/grpc/transport/inprocess/InProcessTransport.java

```diff
@@ -137,18 +138,18 @@ public synchronized void shutdown() {
       return;
     }
     shutdownStatus = Status.UNAVAILABLE.withDescription("transport was requested to shut down");
-    notifyShutdown();
+    notifyShutdown(shutdownStatus);
```

Comment:
Added a comment to ClientTransport:

> If the argument to this function is {@link Status#OK}, it is safe to immediately reconnect.

Assuming shutdown is never called in error, yes.  Done.